crates/subspace-networking/src/request_handlers/piece_by_key.rs

```diff
@@ -5,13 +5,21 @@
 
 use crate::request_handlers::generic_request_handler::{GenericRequest, GenericRequestHandler};
 use parity_scale_codec::{Decode, Encode};
-use subspace_core_primitives::{Piece, PieceIndexHash};
+use subspace_core_primitives::{Piece, PieceIndex, PieceIndexHash};
+
+//TODO: rename all module names
+#[derive(Debug, Clone, Eq, PartialEq, Encode, Decode)]
+pub enum PieceKey {
+    PieceIndex(PieceIndex),
+    PieceIndexHash(PieceIndexHash),
```

Comment:
Pre-hashed value seems native in contrast with converting u64 to bytes when constructing a Multihash (like in multimess file).

But we don't need it functionally because we abandoned the idea of PeerId and PieceIndexHash direct proximity.